controllers/storagecluster/external_resources.go

```diff
@@ -710,7 +710,38 @@ func getTopologyConstrainedPoolsExternalMode(data map[string]string) (string, er
 	return string(topologyConstrainedPoolsStr), nil
 }
 
-func (r *StorageClusterReconciler) enableCsiDrivers(availableSCCs []StorageClassConfiguration) error {
+// getTopologyFailureDomainConfig retrieves the topology failure domain label and values from external resources
+func (r *StorageClusterReconciler) getTopologyFailureDomainConfig(uid types.UID) (string, string, error) {
+	data, ok := externalOCSResources[uid]
+	if !ok {
+		return "", "", fmt.Errorf("unable to retrieve external resource from externalOCSResources")
+	}
+
+	// Look for the topologyFailureDomainLabel and topologyFailureDomainValues in the external resources
+	for _, d := range data {
+		if d.Kind == "StorageClass" && d.Name == cephRbdTopologyStorageClassName {
+			label, labelExists := d.Data["topologyFailureDomainLabel"]
```

Comment:
our validation or defaults doesn't change based on the existence of values, so why can't just do 
```go
label := d.Data["label"]
value := d.Data["value"]
if len(label) > 0 && len(value) > 0 { // use it }
```

Reply:
in the current way it makes it easy to handle error case for validating the inputs from users